File: src/day02.js
import { readFileSync } from "node:fs";

const input = readFileSync(process.stdin.fd, "utf-8").trim();

const idRanges = input.trim().split(',')

let part1 = 0;
let part2 = 0;

for (const range of idRanges) {
    const [start, end] = range.split('-').map(e => +e);

    for (let i = start; i <= end; ++i) {
        const str = "" + i
        if (/^(\d+)\1$/.test(str)) part1 += i;
        if (/^(\d+)\1+$/.test(str)) part2 += i;
    }
}

console.log(part1)
console.log(part2)

File: src/day02.pl
my $part1 = 0;
my $part2 = 0;

while (<>) {
	my @ranges = split(/,/);
	foreach (@ranges) {
		my @nums = split(/-/);
		my $start = @nums[0];
		my $end = @nums[1];

		for($i = $start; $i <= $end; ++$i) {
			if($i =~ /^(\d+)\1$/) {
				$part1 += $i;
			}
			if($i =~ /^(\d+)\1+$/){
				$part2 += $i;
			}
		}
	}
}
print "$part1\n$part2\n";
